orc8r/cloud/go/swagger/combine_test.go

```diff
@@ -0,0 +1,155 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"magma/orc8r/cloud/go/swagger"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/stretchr/testify/assert"
+)
+
+func init() {
+	//_ = flag.Set("alsologtostderr", "true") // uncomment to view logs during test
+}
+
+var (
+	testdataDir    = "testdata"
+	cfgsDir        = filepath.Join(testdataDir, "configs")
+	commonFilepath = filepath.Join(testdataDir, "common/common.yml")
+	outFilepath    = filepath.Join(testdataDir, "out.yml")
+	goldenFilepath = filepath.Join(testdataDir, "out.yml.golden")
+
+	nErrsFromCombine = 5
+)
+
+// TestCombine tests the generated output against a golden file.
+// Golden file was generated by the combine_swagger command, then manually
+// compared to go-swagger's mixin output of the same files, ensuring
+// the golden file's correctness.
+//
+// go-swagger mixin: https://goswagger.io/usage/mixin.html
+func TestCombine(t *testing.T){
+	cleanup()
+	defer cleanup()
+
+	cfgs, common, err := load(cfgsDir, commonFilepath)
+	assert.NoError(t, err)
+
+	out, err, errs := swagger.Combine(common, cfgs)
+	assert.NoError(t, err)
+
+	assert.Error(t, errs)
+	merrs, ok := errs.(*multierror.Error)
+	assert.True(t, ok)
+	assert.Len(t, merrs.Errors, nErrsFromCombine)
+
+	err = write(out, outFilepath)
```

Comment:
This test doesn't need to do any writing -- only reading. The fact that the golden files are stored as files is an implementation detail. That is, though it would be less than ideal, we could just as soon define the golden file inline within this test file

Reply:
Ah I didn't think about that, I was just basing it off the existing test. I forgot that the write to the file doesn't really do much since we are just reading back from it.